community/kernel/src/test/java/org/neo4j/test/assertion/Assert.java

```diff
@@ -41,6 +41,23 @@ private Assert()
     {
     }
 
+    public static <E extends Exception> void assertException( ThrowingAction<E> f, Class typeOfException ) throws
```

Comment:
Also if this method is introduced why not to replace all usages that have dummy expected message passed like in IndexRuleTest: `assertException( indexRule::getOwningConstraint, IllegalStateException.class, "" );` ?